pkg/log/options.go

```diff
@@ -24,15 +24,50 @@ import (
 const (
 	// None is used to disable logging output as well as to disable stack tracing.
 	None zapcore.Level = 100
+
+	defaultOutputLevel        = zapcore.InfoLevel
+	defaultStackTraceLevel    = None
+	defaultOutputPath         = "stdout"
+	defaultErrorOutputPath    = "stderr"
+	defaultRotationMaxAge     = 30
+	defaultRotationMaxSize    = 100 * 1024 * 1024
+	defaultRotationMaxBackups = 1000
 )
 
 // Options defines the set of options supported by Istio's component logging package.
 type Options struct {
 	// OutputPaths is a list of file system paths to write the log data to.
 	// The special values stdout and stderr can be used to output to the
-	// standard I/O streams.
+	// standard I/O streams. THis defaults to stdout.
```

Comment:
Does it make sense to do rotation if you're using stdout/stderr? I would think that the list should either have a single element `stdout` or have a list of files, but not both. Would it make more sense to eliminate the use of `stdout` altogether and just assume `stdout` as the default if the list is empty?   The same would be true for `ErrorOutputPaths` with `stderr`.

Reply:
The default is output to stdout with no log rotation.

If you use log rotation, you must supply explicit output paths. The first such output path cannot be stdout and will be the file that gets rotated. Other output paths will be written to without rotation and can include stdout.

I could simplify things if we want. The multiple output paths thing may be overkill (zap supports it inherently, that's why I exposed it). We could go instead with:

- A boolean to control output to stdout
- A single rotation output path.

Or perhaps:

- A plain single output path. This could be a filename, stdout, or stderr
- A rotation output path. This would be a filename.

WDYT?

Reply:
Oh, I must have read through the code too quickly (apologies) ... I didn't know that log rotation only operated on a single file.  I assumed that it would round-robin across the files in the list, overwriting as necessary.

Let me make sure I understand what we want to do:
- Allow user to specify a log file to be rotated
  - When it's full, future writes will result in losing earliest data.
  - stdout/stderr should never be used
- Allow user to specify at least one non-rotating log file (e.g. stdout)
  - default is stdout/stderr

I would think that maybe the special bit here is the rotating support.  I think what you have for `OutputPaths` is fine, but maybe they shouldn't be involved in rotation at all.  Perhaps we should allow the user to specify a `RotatingOutputPath` (or `RotatingLogFile` since it's specifically a file and not a directory) to enable rotation?

WDYT?

Reply:
First, a correction. Rotation doesn't generally destroy the log file. It renames the current log file by inserting a timestamp into the name and then starts writing to the original name again. You can configure how many of these backups to keep and/or how long to keep old ones.

Second, I think what you're suggesting was my second option from above. A specific path for output (a file, stdout, or stderr) and a specific distinct path for rotation (always a file). Is that right?

Reply:
> First, a correction. Rotation doesn't generally destroy the log file. It renames the current log file by inserting a timestamp into the name and then starts writing to the original name again. You can configure how many of these backups to keep and/or how long to keep old ones.

Ok that makes sense.  So for rotation we'll specify a directory (and maybe a log file prefix), correct? (since we'll be creating files as we go)

> Second, I think what you're suggesting was my second option from above. A specific path for output (a file, stdout, or stderr) and a specific distinct path for rotation (always a file). Is that right?

Yes, basically.  The only difference is that I think the non-rotated paths should allow multiple .. not just one.  That way we can log to multiple sources at once (stdout and a file).